libs/datasets/timeseries.py

```diff
@@ -94,34 +95,42 @@ def get_subset(
         after=None,
         before=None,
         country=None,
-        state=None,
-        county=None,
-        fips=None,
-        states=None,
+        state: Optional[str] = None,
+        county: Optional[str] = None,
+        fips: Optional[str] = None,
+        states: Optional[List[str]] = None,
+        columns_slice=None,
     ) -> "TimeseriesDataset":
-        data = self.data
 
+        # country is okay with states.
+        assert sum([bool(state), bool(county), bool(fips), bool(states)]) <= 1
+
+        query_parts = []
         if aggregation_level:
-            data = data[data.aggregate_level == aggregation_level.value]
+            query_parts.append(f'aggregate_level == "{aggregation_level.value}"')
```

Comment:
I think `LatestValuesDataset` has the same sort of logic here.  Can you update that function as well?

Reply:
Factored out to dataset_utils.